test/models/file_unit_test.rb

```diff
@@ -43,5 +43,18 @@ def test_large_query
       ids = ['href'] * 70_000 + [@file_one.pulp_id]
       assert_equal 1, FileUnit.with_pulp_id(ids).count
     end
+
+    def test_large_sync_repository_association
+      i, ids, ids_href_map = 0, [], {}
+      while (i < 70_000)
+        ids[i] = ["href#{i}"]
+        ids_href_map["href#{i}"] = nil
+        i += 1
+      end
+      Katello::FileUnit.import([:pulp_id], ids)
```

Comment:
Adding  validate: false shaves off about 20 seconds from this method call.  It still takes about ~14 seconds, but i think thats okay.  I don't know that i see a ton of value for having an errata test and a file test, any reason to have the errata test in addition?

Reply:
Right..I dont think we need the validate:true for this. 
The errata test follows a different path where there are actually mappings in the id_href_map..So the 2 flows are the map having nil values for all keys on all content types minus errata and map having actual pulp3_hrefs for errata that we want to store on the katello_repository_erratum table.